src/main/java/com/example/epam/dao/TrainerDao.java

```diff
@@ -0,0 +1,68 @@
+package com.example.epam.dao;
+
+import com.example.epam.entity.Trainer;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class TrainerDao {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public TrainerDao(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    public void save(Trainer trainer) {
+        try (Session session = sessionFactory.openSession()) {
+            session.beginTransaction();
+            session.persist(trainer);
+            session.getTransaction().commit();
```

Comment:
pls, add transaction fallback behavior. 
In case you get an error during execution of the transaction what will be happening.